### PR TITLE
Item 8856: Add tests for merging of list data

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
We allow merging of data for sample types and datasets within the LKS. This change enables that functionality for lists as well.

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/2290

#### Changes
* Add tests for merging of list data
* Update list helper with some more helpful methods

### CODE REVIEW

File: src/org/labkey/test/tests/list/ListTest.java [hunk 1]
```diff
@@ -690,11 +690,67 @@ public void testBatchInsertErrors() throws IOException, CommandException
...
+        checker().verifyTrue("For list with an integer, non-auto-increment key, merge option should be available", _listHelper.isMergeOptionPresent());
+        _listHelper.chooseFileUpload();
+        checker().verifyTrue("For list with an integer, non-auto-increment key, merge option should be available", _listHelper.isMergeOptionPresent());
+        _listHelper.chooseCopyPasteText();
+
+        log("Try to upload the same data without choosing to merge.  Errors are expected.");
+        String[] expectedErrors = new String[]{
+                "duplicate key value violates unique constraint"
+        };
+        setListImportAsTestDataField(toTSV(BatchListColumns, BatchListData), expectedErrors);
```
Comment (Sigmonia) [inline src/org/labkey/test/tests/list/ListTest.java]: Should this error check be done with new data, so a check can be made to ensure no changes were applied?

File: src/org/labkey/test/tests/list/ListTest.java [hunk 2]
```diff
@@ -690,11 +690,67 @@ public void testBatchInsertErrors() throws IOException, CommandException
...
+    {
+        String mergeListName = "autoIncrementIdList";
+
+        _listHelper.createList(PROJECT_VERIFY, mergeListName, ListColumnType.AutoInteger, "Key", col("Name", ListColumnType.String));
+
+        _listHelper.clickImportData();
+        checker().verifyFalse("For list with an integer, auto-increment key, merge option should not be available", _listHelper.isMergeOptionPresent());
+    }
+
+    private void verifyListData(List<ListColumn> columns, String[][] data)
```
Comment (Sigmonia) [inline src/org/labkey/test/tests/list/ListTest.java]: Could this be useful in the listHelper?

**Review by Sigmonia — APPROVED** [review]

**Review by labkey-danield — APPROVED** [review]

File: src/org/labkey/test/tests/list/ListTest.java (hunk 2 above)
Comment (labkey-susanh) [reply src/org/labkey/test/tests/list/ListTest.java]: Yes, good idea.

File: src/org/labkey/test/tests/list/ListTest.java (hunk 1 above)
Comment (labkey-susanh) [reply src/org/labkey/test/tests/list/ListTest.java]: Sure.  Done.

**Pull request merged**